links/src/utils/generateLinks.ts

```diff
@@ -10,8 +10,78 @@ type GenerateLinks = {
  */
 export const generateLinks: GenerateLinks = async () => {
 	const links = await getLinks();
+	const arr = Object.values(links);
+	getSiblings(arr, "cancerqld/research");
+	getParents(arr, "cancerqld/research/viertel-cancer-research-centre");
+	return;
+};
+
+export type Node = {
+	id: number;
+	parent_id: number;
+	slug: string;
+};
+
+export type NodeWithChildren = Node & { children?: NodeWithChildren[] };
+
+export type Tree = NodeWithChildren[];
+
+export type ToTree = {
+	(arr: Node[], parent_id?: number): Tree;
+};
 
-	// TODO: logic
+export type GetSiblings = {
+	(arr: Node[], slug: string): Tree;
+};
+
+// getSiblings(tree, '/cancerqld/research')
+export const getSiblings: GetSiblings = (arr, slug) => {
+	let tree: Tree = [];
+	let parent: number;
+	for (const node of arr) {
+		if (node.slug === slug) {
+			parent = node.parent_id;
+			tree = toTree(arr, parent);
+		}
+	}
+	return tree;
+};
+export const toTree: ToTree = (arr, parent_id = 0) => {
+	let tree: Tree = [];
+	for (const node of arr) {
+		// use key value to get parent
+		if (node.parent_id === parent_id) {
+			let child_tree = toTree(arr, node.id);
+			let children = child_tree.length ? child_tree : undefined;
+			tree.push({ ...node, children });
+		}
+	}
+	return tree;
+};
+
+// getParents(tree, '/cancerqld/research') get parent_id of slug
+export const getParents: GetSiblings = (arr, slug) => {
```

Comment:
Where is GetParents?

```suggestion
export const getParents: GetParents = (arr, slug) => {
```